pkg/pwd/pwd_cgo.go

```diff
@@ -0,0 +1,56 @@
+//go:build !windows
+
+/*
+Package pwd is a thin wrapper of C library <pwd.h>.
+This is designed as thin as possible, but aimed to be thread-safe.
+*/
+package pwd
+
+/*
+#include <sys/types.h>
+#include <pwd.h>
+#include <stdlib.h>
+
+// While getpwuid requires "uid_t" according to man page, it actually requires
+// "__uit_t" in the source code, that causes cgo compile error ("uid_t" is
+// actually aliased to __uid_t).
+// Unlike Linux, getpwuid on Mac OS X requires uid_t properly. For compatibility,
+// we use a C function as a bridge here.
+struct passwd *getpwuid_aux(unsigned int uid) {
+	return getpwuid((uid_t)uid);
+}
+*/
+import "C"
+
+// Passwd represents an entry of the user database defined in <pwd.h>
+type Passwd struct {
+	Name   string // user name
+	Passwd string // user password
+	UID    uint32 // user ID
+	GID    uint32 // group ID
+	Gecos  string // real name
+	Dir    string // home directory
+	Shell  string // shell program
+}
+
+func cpasswd2go(cpw *C.struct_passwd) *Passwd {
```

Comment:
Why do we need a private func called only once and that only returns a pointer to a struct?

Reply:
Removed 👍